### PR TITLE
Update cask naming spec and rename fonts to match

### LINKED ISSUE #521: Draft naming spec and rename
Per the discussion that started in #515, I think the old seat-of-the-pants naming was working out better that the letter-of-the-law names without hyphens recently kludged in from the cask naming standards.

I propose _not_ pretending font file names are the same as Mac app names "as found on disk" and using the font face name as specified by the font itself instead.

- [x] Document how font casks should be named (documentation existed in CONTRIBUTING.md, it just needed tweaking)
- [x] Rename existing font casks to match (recently renamed fonts that don't comply reverted, others to be reset on an as-spotted bases)

### PR DESCRIPTION
**Pull request by alerque**
Fixes #521.

### CODE REVIEW

**Comment by vitorgalvao** [comment]
Thank you.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -81,7 +81,7 @@ predictable.
 ### Start From the Font's Canonical Name
 
 The canonical font name is the font family name as returned by the command
-`otfinfo --family`.
+`fc-query -f '%{family}'`.
```
Comment (vitorgalvao) [inline CONTRIBUTING.md]: Long flags are always better for documentation and scripts. Do instead `fc-query --family '%{family}' <pat_to_font_file>`.

Comment (alerque) [inline CONTRIBUTING.md]: @vitorgalvao It's not `--family` it's `--format`, but I'll replace it with the long option.

**Comment by alerque** [comment]
Edited to use long-format option syntax...

**Pull request merged**